cpp/src/join/joining.cu

```diff
@@ -533,6 +534,70 @@ gdf_error join_call_compute_df(
     return GDF_SUCCESS;
   }
 
+
+  //if the inputs are nvcategory we need to make the dictionaries comparable
+  gdf_column ** new_left_cols = new gdf_column * [num_left_cols];
```

Comment:
First of all, you should only need to perform this check for the `left_join_cols` and `right_join_cols`, not all of `left_cols` and `right_cols`. There's no need to synchronize category columns if they're not being joined on. 

Second of all, use a `std::vector<gdf_column*>` here instead. You can also avoid the `for` loops below.

```
std::vector<gdf_column*> new_left_join_cols(left_join_cols, left_join_cols + num_cols_to_join);
```